Lib/test/test_hashlib.py

```diff
@@ -1196,6 +1196,29 @@ def test_file_digest(self):
             with open(os_helper.TESTFN, "wb") as f:
                 hashlib.file_digest(f, "sha256")
 
+    @unittest.skipUnless(support.check_sanitizer(thread=True), "only meaningful on free-threading")
```

Comment:
- IIRC, TSan is run for the default build as well, because things like subinterpreters can have thread safety issues.
- I think it's worth always running this for FT, because we'll need to assert that the thread doesn't raise exceptions (more on that in another comment).
- Should only run if threading actually works.
```suggestion
    @unittest.skipUnless(support.Py_GIL_DISABLED, "only meaningful on free-threading")
    @threading_helper.requires_working_threading()
```